arangod/Replication2/ReplicatedLog/Components/StorageManager.cpp

```diff
@@ -447,11 +443,16 @@ auto StorageManager::getCommittedLogIterator(
 }
 
 auto StorageManager::getSyncIndex() const -> LogIndex {
-  return syncIndex.getLockedGuard().get();
+  return syncIndex.getLockedGuard()->index;
 }
 
-void StorageManager::updateSyncIndex(LogIndex index) {
+void StorageManager::updateSyncIndex(TermIndexPair index) {
   syncIndex.doUnderLock([index](auto& currentIndex) {
+    TRI_ASSERT(index.term >= currentIndex.term)
+        << "trying to update sync "
+           "index with a lower term "
+           "than the current one "
+        << currentIndex << " vs " << index;
```

Comment:
This can and will fail in production. It is totally valid for the leader to send log entries that have a lower term than the current term. Those log entries might be entirely new to the follower and when they are synced, this assertion will simply pop.

Reply:
I have reverted this